toolchain/check/impl_lookup.cpp

```diff
@@ -340,8 +340,8 @@ static auto FindWitnessInImpls(Context& context, SemIR::LocId loc_id,
                                SemIR::ConstantId query_self_const_id,
                                SemIR::SpecificInterface interface)
     -> SemIR::InstId {
-  auto witness_const_id = TryEvalInst(
-      context, loc_id, SemIR::InstId::None,
+  auto witness_id = AddInstInNoBlock(
```

Comment:
Maybe we could document then how this instruction ends up with an id in the symbolic LookupImplWitness case. I guess it is because in that case LookupImplWitness evaluates to itself, so then we have a constant value added to the store, and its instruction is this instruction, so that causes the id to be added? It's a bit subtle.

Reply:
Looking at toolchain/check/testdata/facet/min_prelude/convert_facet_value_to_narrowed_facet_type.carbon as an example, what I'm seeing is that the instruction we create here doesn't appear in the resulting IR. We do have an instruction in the constants block, and an instruction in the eval block (which is produced by substituting into the instruction in the constants block), but the one created here is not visible.

I'll send you a PR that switches to an `EvalOrAddInst` approach here, as you suggested on Discord.

Reply:
Oh I see. The constant value can still be produced (which has an inst within) and that appears in the semir and is referenced accordingly. The inst that created that constant value is lost and unused. Okay, thanks. :)